extensions/example-extension/renderer.tsx

```diff
@@ -9,17 +9,39 @@ export default class ExampleExtension extends LensRendererExtension {
       title: "Example Extension",
       components: {
         Page: () => <ExamplePage extension={this}/>,
+      },
+      params: {
+        // setup param "exampleId" with default value "demo"
+        // could be also {[paramName: string]: UrlParam} for advanced use-cases (custom parse/stringify)
+        exampleId: "demo"
       }
```

Comment:
I find there are conflicting intentions here vs `Navigation.createPageParam(...)` in order to get the param because both places allow you to specify a default value.

So maybe it's cleaner to remove setting the default in `Navigation.getPageParam()` (what I think we should rename it to) and let the default be specified here, and then the way to _get_ the param's value becomes:
`const param = Navigation.getPageParam({ pageId: 'example', param: 'exampleId' });`

Reply:
Nope, all good here. You can already define param like here:
```
params: {
  exampleId: Navigation.createPageParam(...)
}
```
But renaming might be a good idea.

Reply:
😮  Too many ways to use the same thing for different purposes... If you can do this, then we should probably keep it (and let it accept a defaultValue), but still also provide the reader, `Navigation.getPageParam(pageId, paramId)` to get the same object out of the thingamagig (another "store"?) that stores these parameters when you want to get the param instead of declare the param.

And this is optional, right? If I don't care about observability, I can just use a static default value?
```javascript
params: {
  exampleId: 'foo'
}
```

Reply:
This part of code from where? ^ What is optional? 
Declaring params is non-optional for page registration (if page need them it should know how to process them).

Reply:
+ to avoid passing all unknown params  to menus for example (remember: extension's cross-linking is possible)

Reply:
> And this is optional, right? If I don't care about observability, I can just use a static default value?

and if you don't care about param type value while you do `param.get()` (otherwise you can setup custom parsing/stringification logic), otherwise just use plain `string` for default-value as in your code above.

Reply:
I'm sorry I wasn't clear enough. What I meant to ask was if there are 3 supported ways of declaring params for a page:
```javascript
params: {
  foo: 'foo-default',
  bar: Navigator.createPageParam({ pageId: 'example', paramId: 'bar', defaultValue: 'bar-default' }),
  baz: ... // this other parsing/stringification thing you've mentioned, which I don't know about
}
```

(It feels a bit weird to use `Navigator.createPageParam()` here, almost too heavy-handed given that even if I don't use it, as in the 'foo' param, I can then still use `Navigator.getPageParam()` to get the same type of observable object for it. So I wouldn't be sad to see support for this dropped. I don't think @jim-docker would be sad either, based on the simplicity of his proposal in #1399)

And then to consume/receive those params, there are 2 ways:

First, in `renderer.tsx` as props from the page declaration (doesn't work yet, but you will add):
```javascript
//// renderer.tsx

...
clusterPages = [
    {
      id: "example",
      title: "Example Extension",
      components: {
        Page: ({ params }) => <ExamplePage extension={this} {...params} />,
      },
      params: {...} // the object from above
    }
]
```

Second, in `page.tsx` using the new `Navigation.getPageParam(pageId, paramId)` (doesn't exist yet, but you will hopefully add).
```javascript
//// page.tsx

const fooParam = Navigation.getPageParam('example', 'foo');

export function MyPage() {
  return <p>{fooParam.get()}</p>;
}

// or I could receive as props per my declaration example:
export function MyPage({ foo, bar, baz }) {
  return <p>{foo} {bar.get()} {baz???}</p>;
}
```

Reply:
Currently only 2 ways of declaring params for the page:

```
params: {
 // basic: setup param-name and default value, converted anyway to `PageParam` object when provided to Page component in `props.params` == aka "normalized params"
  foo: 'foo-default',

 // advanced (via direct `PageParam` helper call): supports custom stringify/parse/multi-values
  bar: Navigator.createPageParam(...), 
}
```

Reply:
Got it, so `bar` and `baz` are actually the same thing in my examples. I thought there was a third case. Thanks for clarifying.